skimage/draw/draw.py

```diff
@@ -163,7 +166,38 @@ def set_color(img, coords, color):
            [0, 0, 0, 0, 0, 0, 0, 0, 0, 1]], dtype=uint8)
 
     """
-
     rr, cc = coords
-    rr, cc = _coords_inside_image(rr, cc, img.shape)
-    img[rr, cc] = color
+
+    if alpha is None:
+        rr, cc = _coords_inside_image(rr, cc, img.shape)
+    else:
+        rr, cc, alpha = _coords_inside_image(rr, cc, img.shape, val=alpha)
+
+    if not np.isscalar(color):
+        color = np.array(color)
```

Comment:
I would use `np.asanyarray` or `np.asarray` here, so that arrays are not copied unnecessarily. It's tiny performance gain here but I think it's useful to maintain proper usage examples throughout our codebase.